### PR TITLE
Find Water Source Mission

### LINKED ISSUE #99: Mission Creation: Find Water Mission
### Utilize Sensors to Find Underground Water

**Unity Objects & Assets:**
- Create a sensor object (3D model of a handheld device or stationary unit).
- Add invisible WaterSource prefabs to mark underground water locations using colliders.
- Design a UI element (e.g., a radar or proximity meter) to show sensor feedback.

**SensorScript:**
Detect water sources using colliders/triggers (e.g., Physics.OverlapSphere).
Provide visual/audio cues when water is detected.
Update UI to guide the player.
Integrate into MissionManager to track progress.

**Find Explosives**
- Add an explosive object (e.g., crate or dynamite model). Use a placeholder if necessary.
- Implement interactable markers or zones to collect explosives.

PickupScript: Allow players to collect explosives and update mission state in MissionManager.

**Blow Up Area to Uncover Water**

Design a destructible terrain/object representing the explosion site (e.g., rocks, dirt pile).
Use a particle system for explosion effects (fire, smoke, debris).

ExplosionScript: Execute explosion, destroy covering object, and reveal the water fissure.
DestructionManager: Manage terrain/object changes.
Reveal Water Fissure

Create or import a water fissure prefab (particle system with flowing water and steam effects).
Add terrain decals for cracks or fissures.
 Scripts:
RevealFissureScript: Activate the water fissure prefab and update mission progress in MissionManager.

Assets Needed
 Explosives: Obtain dynamite/TNT crate models (e.g., from Unity Asset Store).
 Explosion Effects: Find/create particle effects for fire, smoke, and debris.
 Water Fissure: Use a particle system for flowing water.
 Terrain Features: Create destructible rocks or dirt piles.

### PR DESCRIPTION
**Pull request by aamirnaje**
### **Summary**
This pull request introduces a new mission, Find Water Source, which includes multiple sub-missions and significant gameplay enhancements. Players will now experience a more immersive and challenging journey to locate and access a water source. This update involves changes to existing scripts, the addition of new scripts and assets,

### **Key Features**

**New Sub-Missions**

1. Find GPS Tracker:  Players must locate a GPS tracker, which enables the radar functionality to find water
2. Find Dynamite:** : Players are tasked with finding dynamite to progress further in the mission.
3. Locate the Water Cave Entrance:**  The radar directs players to the entrance of the water cave.
4. Blow Up the Water Cave Door:** : Players use the dynamite to destroy the entrance to the water cave.

_**DynamiteItem.cs**_

- Handles all mechanics for dynamite interactions, including:
- Detonation effects.
- Interaction with destructible objects like the water cave door.
- Spawning mini blocks to simulate realistic debris.
- Dynamically integrates with the mission log to track and update mission progress.

_**WaterQuest.cs**_

- Manages the final stage of the mission, marking the water source as found and completing the quest.

_**Radar Asset**_

- A functional radar system that:
  - Displays the player’s location in relation to the GPS-marked water source.
  - Activates only after the GPS scanner item is picked up by the player.

**Existing Script Enhancements**

**GameState Script**

- Updated battery mechanics to decrease battery level when the player is near a detonation.

closes #99

### CODE REVIEW

File: Assets/Mirza Beig/Cinematic Explosions FREE/Animations/Explosive Force Field.anim
Comment (Amal-Mathew204) [inline Assets/Mirza Beig/Cinematic Explosions FREE/Animations/Explosive Force Field.anim:1]: If this is the old explosion please remove

File: Assets/New Terrain.asset
Comment (Amal-Mathew204) [inline Assets/New Terrain.asset:1]: why two terrian assests. Please check this.

File: Assets/Scenes/AamirScene.unity
Comment (Amal-Mathew204) [inline Assets/Scenes/AamirScene.unity:1]: Remove Personal Scenes

File: Assets/Scripts/Game/GameManager.cs [hunk 1]
```diff
@@ -216,6 +233,24 @@ public void ChangeEnterTextFieldVisibility(bool visibility)
...
+
+        #region Game State Methods
+        /// <summary>
+        /// 
+        /// </summary>
+        public void SetBatteryLevelReduction(float percentageReduction)
+        {
+            _gameState.SetBatteryLevelReduction(percentageReduction);
+        }
+
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Game/GameManager.cs:245]: missing summary

File: Assets/Scripts/Game/GameState.cs [hunk 2]
```diff
@@ -57,5 +57,22 @@ private void SetBatteryLevel(int level)
...
+            if(percentageReduction <= 0 &&  percentageReduction > 1)
+            {
+                Debug.LogError("Invalid Percentage Reduction");
+                return;
+            }
+            float gameTimeLeft = (_batteryLevel / 100f) * _totalGameTime;
+            _gameElapsedTime += gameTimeLeft * percentageReduction;
+        }
+
+
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Game/GameState.cs:76]: missing summary

File: Assets/Scripts/Item/DynamiteItem.cs
Comment (Amal-Mathew204) [inline Assets/Scripts/Item/DynamiteItem.cs:1]: Use Regions in this file

File: Assets/Scripts/Item/GPSScannerPickup.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                    gameObject.SetActive(false);
+
+                    if(itemName == "GPS Scanner" && radar != null)
+                    {
+                        PlayerCharacter.Instance.radar.ActivateRadar();
+                       
+                    }
+                }
+            }
+        }
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Item/GPSScannerPickup.cs:34]: Missing Summary

File: Assets/Scripts/Player/Player.cs [hunk 4]
```diff
@@ -13,6 +13,8 @@
 using Scripts.Quests;
 using Scripts.NPC;
 using MissionLogDropdown = Scripts.MissonLogMenu.Dropdown;
+using Ilumisoft.RadarSystem;
+using UnityEditor.Rendering;
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Player/Player.cs:17]: Add comment that `using Ilumisoft.RadarSystem;` is a third party class

File: Assets/Scripts/Player/PlayerLocatable.cs
Comment (Amal-Mathew204) [inline Assets/Scripts/Player/PlayerLocatable.cs:1]: File has no functionality here. If this is not important remove

File: Assets/Scripts/Quests/WaterQuest.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        private MissionLogDropdown _dropdown;
+
+        /// <summary>
+        /// Initializes the script by finding the Mission UI component and assigning it to the dropdown variable.
+        /// This is called when the script starts.
+        /// </summary>
+        public void Start()
+        {
+            _dropdown = GameObject.FindGameObjectWithTag("MissionUI").GetComponent<MissionLogDropdown>();
+        }
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Quests/WaterQuest.cs:19]: Use `GameManager.cs` method to get `MissionLogDropdown` component

**Review by Amal-Mathew204 — CHANGES_REQUESTED** [review]

File: Assets/Scenes/AamirScene.unity
Comment (aamirnaje) [reply Assets/Scenes/AamirScene.unity:1]: removed

File: Assets/Scripts/Game/GameManager.cs (hunk 1 above)
Comment (aamirnaje) [reply Assets/Scripts/Game/GameManager.cs]: summary added

File: Assets/Scripts/Item/DynamiteItem.cs
Comment (aamirnaje) [reply Assets/Scripts/Item/DynamiteItem.cs:1]: regions added

File: Assets/Scripts/Player/Player.cs (hunk 4 above)
Comment (aamirnaje) [reply Assets/Scripts/Player/Player.cs:17]: commented usage of third party class

File: Assets/Scripts/Player/PlayerLocatable.cs
Comment (aamirnaje) [reply Assets/Scripts/Player/PlayerLocatable.cs:1]: removed

**Review by Amal-Mathew204 — APPROVED** [review]
Best PR to date

**Pull request merged**